js/components/graph/Graph.js

```diff
@@ -425,73 +517,20 @@ export default class Graph extends React.Component {
     return Math.max(heightToContainerRatio, widthToContainerRatio);
   };
 
-  panDirection = (direction, panFactorRate) => {
-    // onButtonRelease calls are required when a button becomes disabled
-    // because it loses its ability to detect mouseUp event
-    if (direction === "up") {
-      this.setState({
-        verticalPanFactor: this.state.verticalPanFactor - panFactorRate
-      });
-    } else if (direction === "left") {
-      this.setState({
-        horizontalPanFactor: this.state.horizontalPanFactor - panFactorRate
-      });
-    } else if (direction === "down") {
-      this.setState({
-        verticalPanFactor: this.state.verticalPanFactor + panFactorRate
-      });
-    } else if (direction === "right") {
-      this.setState({
-        horizontalPanFactor: this.state.horizontalPanFactor + panFactorRate
-      });
-    }
-  };
-
   resetZoomAndPan = () => {
+    this.zoomViewbox(this.zoomEnum.RESET_ZOOM);
```

Comment:
What about setting `zoomFactor` to 1, and resetting `viewBoxDim` to it's original value in this call to `setState` and remove the call to `zoomViewBox`? So that would look like
```
this.setState({
zoomFactor:1,
verticalPanFactor:0,
horizontalPanFactor:0,
viewBoxDim: {width: this.state.width, height: this.state.height}
```

It would mean `zoomViewbox` isn't the only location where the viewbox's dimensions are changed, but I think it is fine. Let me know what you think.

Reply:
Yes, I think that could work as well.